src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -158,6 +162,8 @@ namespace cryptonote
       std::string status;
       std::vector<block_output_indices> output_indices;
       bool untrusted;
+      uint64_t credits;
+      std::string top_hash;
```

Comment:
Is every RPC call including this to make it easier for the pay-for-RPC case? Is there some technique where this an just be inserted to every response? Or can this be dropped (efficiency)? I mean its adding to lots of changes and Im not entirely sure why.

Reply:
All these are for the pay-for-rpc system, yes. credits allows the user to know what was subtracted, top_hash is to know if it should update its hashing blob if a block was found. I think this might be simplified with some system you typed out above (which was for signing the entire system, a similar thing might be usable for these two fields too). If you see a better way, I'm all ears, I don't really like the spam either.